.github/workflows/psm-interop.yaml

```diff
@@ -0,0 +1,74 @@
+name: PSM Interop
+
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+      - 'v1.*'
+
+permissions:
+  contents: read
+
+jobs:
+  unittest:
+    # By default, only version is printed out in parens, e.g. "unittest (3.10)"
+    # This changes it to "unittest (python3.10)"
+    name: "unittest (python${{ matrix.python_version }})"
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python_version: ["3.9", "3.10"]
+      fail-fast: false
+    permissions:
+      pull-requests: read  # Used by paths-filter to read the diff.
+    defaults:
+      run:
+        working-directory: 'tools/run_tests/xds_k8s_test_driver'
+
+    steps:
+      - uses: actions/checkout@v3
+
+      # To add this job to required GitHub checks, it's not enough to use
+      # the on.pull_request.paths filter. For required checks, the job needs to
+      # return the success status, and not be skipped.
+      # Using paths-filter action, we skip the setup/test steps when psm interop
+      # files are unchanged, and the job returns success.
+      - uses: dorny/paths-filter@v2
+        id: paths_filter
+        with:
+          filters:  |
+            psm_interop_src:
```

Comment:
Fair enough, but filters like this tend to be hard to fragile and easily become out of date, so the question is whether the timesaving here is worth it (the test is currently very fast, right?).
If this was a bazel tests, you'd be able to make the test hermetic by only depending on the selected files, at which point you could be sure that the path filter always stays up-to-date.

Not blocking on this, but leaving up to you to consider.

Reply:
Yes, good point. I didn't like it much too. For now, I think the benefits outweigh the risks. I _absolutely_ don't want to block anything in the repo because, f.e. newer pip breaks things.

For the long-term, I have plans to move the framework out to its own repo, but we should discuss it separately.